controller.py

```diff
@@ -0,0 +1,103 @@
+import typing
+
+import serial
+import enum
+import logging
+
+import util
+from state import State, Error
+
+
+LOGGER = logging.getLogger('controller')
+
+class StateKeys:
+    CURR_X = 'curr_x'
+    TRGT_X = 'trgt_x'
+    CURR_V = 'curr_v'
+    TRGT_V = 'trgt_v'
+    CURR_A = 'curr_a'
+    TRGT_A = 'trgt_a'
+    POLE_ANG = 'pole_ang'
+    POLE_VEL = 'pole_vel'
+    TIMESTAMP = 'timestamp'
+    ERRCODE = 'errcode'
+
+
+class ActionType(enum.Enum):
+    PositionControl = 1
+    VelocityControl = 2
+    AccelerationControl = 3
+
+
+class Action:
+    def __init__(self, action_type: ActionType, value: float) -> None:
+        self.type = action_type
+        self.value = value
+
+    def __str__(self) -> str:
+        action_str = {
+            ActionType.PositionControl: StateKeys.TRGT_X, 
+            ActionType.VelocityControl: StateKeys.TRGT_V,
+            ActionType.AccelerationControl: StateKeys.TRGT_A,
+            }[self.type]
+        return f'{action_str}={self.value}'
+
+
+class CartPoleController:
+    def __init__(self, device: str, baud_rate: int, step_n: int) -> None:
```

Comment:
1. device -> serial_port,
2. give reasonable defaults

Reply:
Also add as ads timeout for write/read

Reply:
fixed